build.py

```diff
@@ -453,7 +453,7 @@ def update(args: argparse.Namespace, env: dict[str, str]):
 
             is_x64_host = platform.machine() == "AMD64"
             if is_x64_host:
-                toolset_options += ["host=x64"]
+                pass
```

Comment:
Again, this makes `build.py` align with `cmake --preset`.

Reply:
Yes. For our build environment, x64 is the default. I'm guessing this is from ORT when x86 is still a target. For GenAI we only ever target x64.